components/script/dom/gpuqueue.rs

```diff
@@ -191,4 +195,39 @@ impl GPUQueueMethods for GPUQueue {
 
         Ok(())
     }
+
+    /// <https://gpuweb.github.io/gpuweb/#dom-gpuqueue-onsubmittedworkdone>
+    fn OnSubmittedWorkDone(&self) -> Rc<Promise> {
+        let global = self.global();
+        let promise = Promise::new(&global);
+        let sender = response_async(&promise, self);
+        if let Err(e) = self.channel.0.send((
+            self.device.borrow().as_ref().unwrap().use_current_scope(),
+            WebGPURequest::QueueOnSubmittedWorkDone {
+                sender,
+                queue_id: self.queue.0,
+            },
+        )) {
+            todo!("QueueOnSubmittedWorkDone failed with {e}")
```

Comment:
Hrm. `todo!` is an interesting choice here. Why not `warn!` which won't panic?

```suggestion
            warn!("QueueOnSubmittedWorkDone failed with {e}")
```

Reply:
Todo as actually to-do in the future. But now that I look over at other similar code warn would really be more appropriate.